java/org/apache/catalina/session/ManagerBase.java

```diff
@@ -684,6 +714,12 @@ public Session createSession(String sessionId) {
             id = generateSessionId();
         }
         session.setId(id);
+
+        if (session instanceof StandardSession) {
+            // Set persistAuthentication only for StandardSession instances
+            ((StandardSession) session).setPersistAuthentication(persistAuthentication);
```

Comment:
I think this has been discussed on the mailing list. Is there a reason to "bless" each session instead of checking the session manager at the time of serialization?

Reply:
I will commit/push to the branch when most of the already discussed issues are solved.